src/pages/commits/ActiveCommits/buildVariantCard/WaterfallTaskStatusIcon.tsx

```diff
@@ -39,24 +40,23 @@ export const WaterfallTaskStatusIcon: React.VFC<
   const { testResults, filteredTestCount } = data?.taskTests ?? {};
   const failedTestDifference = filteredTestCount - (testResults ?? []).length;
 
-  let timeout;
-  const onMouseEnter = () => {
-    injectGlobalStyle(identifier);
-    timeout = setTimeout(() => {
-      setEnabled(true);
-      // Only query failing test names if the task has failed.
-      if (isFailedTaskStatus(status)) {
-        loadData();
-      }
-    }, 500);
-  };
-  const onMouseLeave = () => {
-    removeGlobalStyle();
-    setEnabled(false);
+  useEffect(() => {
     if (timeout) {
       clearTimeout(timeout);
     }
-  };
+    if (enabled) {
+      injectGlobalStyle(identifier);
+      timeout = setTimeout(() => {
+        // Only query failing test names if the task has failed.
```

Comment:
```suggestion
      timeout = setTimeout(() => {
         injectGlobalStyle(identifier);
        // Only query failing test names if the task has failed.
```

Reply:
My implication was that since injectingGlobalStyle is slow it might make sense to throttle it to only if a user focuses on a task. But you are right it does lead to a worse experience. So I'm ok with your suggestion!